delegate/http.go

```diff
@@ -191,7 +192,15 @@ func (p *HTTPClient) Acquire(ctx context.Context, delegateID, taskID string) (*c
 func (p *HTTPClient) SendStatus(ctx context.Context, delegateID, taskID string, r *client.TaskResponse) error {
 	path := fmt.Sprintf(taskStatusEndpoint, taskID, delegateID, p.AccountID)
 	req := r
-	_, err := p.retry(ctx, path, "POST", req, nil, createBackoff(ctx, taskEventsTimeout), true) //nolint: bodyclose
+	retryNumber := 0
```

Comment:
@devkimittal sounds like what you really want is to put a timeout on the actual API call. You can do that by sending in a context with timeout `x` to `SendStatus` - for example: https://github.com/harness/dlite/blob/main/poller/poller.go#L109.

This means that each API call will take a max of `x` time. And the timeout set in this file will ensure it keeps retrying for a max `y` amount of time. So the API is guaranteed to run `y/x` times.

Reply:
Done, added timeout of 10 seconds per request.